bin/manifest.py

```diff
@@ -0,0 +1,244 @@
+
+def base_decode(s, digits, check = 1):
+    """ Returns the binary from decoding a radix-encoded string """
+    bn = 0
+    radix = len(digits)
+    ls = s.lstrip(str(digits[0]))
+    for c in ls:
+        bn = bn * radix + digits.index(c)
+    b = ''
+    while bn > 0:
+        bn, m = divmod(bn, 256)
+        b += chr(m)
+    b = b'\0' * (len(s) - len(ls)) + b[::-1]
+    if (check):
+        assert base_encode(b, digits, 0) == s
+    return b
+
+def base58encode(b):
+    return base_encode(b, human_alphabet)
+
+def base58decode(b):
+    return base_decode(b, human_alphabet)
+
+
+#
+# RippleAddress library
+#
+# Human strings are base-58 with a
+# version prefix and a checksum suffix.
+#
+
+VER_NONE              = 1
+VER_NODE_PUBLIC       = 28
+VER_NODE_PRIVATE      = 32
+VER_ACCOUNT_ID        = 0
+VER_ACCOUNT_PUBLIC    = 35
+VER_ACCOUNT_PRIVATE   = 34
+VER_FAMILY_GENERATOR  = 41
+VER_FAMILY_SEED       = 33
+
+def ra_check(b):
+    """ Returns a 4-byte checksum of a binary """
+    return sha256(sha256(b).digest()).digest()[:4]
+
+def ra_encode(ver, b):
+    """ Encodes a binary as human string """
+    b = chr(ver) + b
+    return base58encode(b + ra_check(b))
+
+def ra_decode(s):
+    """ Decodes a human base-58 string into its version and binary """
+    b = base58decode(s)
+    check = b[-4:]
+    if (check != ra_check(b[:-4])):
+        raise ValueError('bad checksum')
+    ver = ord(b[0])
+    b = b[1:-4]
+    return ver, b
+
+def field_code(kind, name):
+    s = ""
+    if (kind < 16):
+        if (name < 16):
+            s += chr((kind << 4) + name)
+        else:
+            s += chr(kind << 4)
+            s += chr(name)
+    elif (name < 16):
+        s += chr(name)
+        s += chr(kind)
+    else:
+        s += '\0'
+        s += chr(kind)
+        s += chr(name)
+    return s
+
+#-----------------------------------------------------------
+
+STI_UINT32          = 2
+STI_VL              = 7
+
+sfSequence          = field_code(STI_UINT32, 4)
+sfPublicKey         = field_code(STI_VL, 1)
+sfSigningPubKey     = field_code(STI_VL, 3)
+sfSignature         = field_code(STI_VL, 6)
+
+def to_bytes(n, length, endianess='big'):
+    h = '%x' % n
+    s = ('0'*(len(h) % 2) + h).zfill(length*2).decode('hex')
+    return s if endianess == 'big' else s[::-1]
+
+def lenvl(b):
+    s = ''
+    n = len(b)
+    if (n < 192):
+        s += chr(n)
+        return s
+    raise Exception('too long')
+
+def strvl(b):
+    return lenvl(b) + b
+
+def str32(n):
+    return to_bytes(n, 4)
+
+#-----------------------------------------------------------
+
+def gen_seed():
+    seed = os.urandom(16)
+    return seed
+
+def gen_ed():
+    sk = os.urandom(32)
+    pk = ed25519.publickey(sk)
+    return sk, pk
+
+def gen_ec():
+    sk = ecdsa.SigningKey.generate(curve=ecdsa.SECP256k1)
+    vk = sk.get_verifying_key()
+    sig = sk.sign("message")
+    assert vk.verify(sig, "message")
+    return sk, vk
+
+def gen_manifest(pk, vpk, seq):
+    s = ""
```

Comment:
I grant you `'foo'` vs. `"foo"`, but `''` can be confused for `' '`, a problem not shared with `""` and `" "`